### PR TITLE
Single multi version store

### PR DESCRIPTION
**Pull request by jamesgorrie**
Adds two implementation for the different ways the library can work.

You can either use a hashed ranged store, or just a hash store, and the infrastructure needs to reflect this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,10 +7,23 @@ Creates the storage infrastructure required to use an instance of the HybridStor
 
 ## Usage
 
+You can either use a `hash-store` or `hash-range-store` implementation, which must
+match the type of store you're using in your application.
```
Comment (alexwlchan) [inline README.md:11]: Maybe worth explaining a bit more about what the difference is? e.g. “you can store multiple versions in one, only the latest version in the other”

File: hash-store/iam_policy_documents.tf [hunk 2]
```diff
@@ -0,0 +1,68 @@
+data "aws_iam_policy_document" "read_policy" {
+  # This is based on the AmazonDynamoDBReadOnlyAccess
```
Comment (alexwlchan) [inline hash-store/iam_policy_documents.tf:2]: Is this sentence missing a last word?

**Review by alexwlchan — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jamesgorrie) [reply README.md:11]: Would this explanation best live in the library and we link through to there?

File: hash-store/iam_policy_documents.tf (hunk 2 above)
Comment (jamesgorrie) [reply hash-store/iam_policy_documents.tf:2]: I suppose it's inferred from the type of thing this is. Added the word policy.
https://github.com/wellcometrust/terraform-modules/blob/master/vhs/modules/vhs/iam_policy_document.tf#L2

**Comment by jamesgorrie** [comment]
@alexwlchan merging, if we think the explanation of the different types are worth having here rather than in the library to link to, I will do that after.

**Pull request merged**